component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/record/SchemaImpl.java

```diff
@@ -132,6 +235,33 @@ public Builder withEntry(final Entry entry) {
                 this.entries.add(entryToAdd);
             }
 
+            entriesOrder.add(entry.getName());
+            return this;
+        }
+
+        @Override
+        public Builder withEntryAfter(final String after, final Entry entry) {
+            if (entriesOrder.indexOf(after) == -1) {
+                throw new IllegalArgumentException(String.format("%s not in schema", after));
+            }
+            withEntry(entry);
+            entriesOrder.remove(entry.getName());
+            int destination = entriesOrder.indexOf(after);
+            if (!(destination + 1 == entriesOrder.size())) {
+                destination += 1;
+            }
+            entriesOrder.add(destination, entry.getName());
+            return this;
+        }
+
+        @Override
+        public Builder withEntryBefore(final String before, final Entry entry) {
+            if (entriesOrder.indexOf(before) == -1) {
```

Comment:
direct call to this.moveBefore(before, entry.getName()) could be more readable ?

Reply:
Sure!